### ISSUE TITLE
Windows: Create exe with one name & use another for the add/remove programs list

### ISSUE THREAD

**Issue by warrenbuckley** [electron-userland/electron-builder#2813]
* **Version**: 
`"electron-builder": "^20.9.0"`

* **Target**: 
Windows: NSI

Hello,
I am after some help in creating my Electron application for windows using NSI which creates the installer for me.

I wish to have Electron to produce the exe as one name and to have a different name used for the add/remove programs uninstall list.

Here is a copy of my build section from my package.json
```
"build": {
    "appId": "lucene.index.explorer",
    "productName": "Lucene Index Explorer",
    "extraResources": {
      "from": "../LIE.Server/bin/dist/",
      "to": "LIE.Server/bin/dist/",
      "filter": [
        "**/*"
      ]
    },
    "win": {
      "target": [
        "nsis"
      ],
      "icon": "build/icon.png"
    },
    "nsis": {
      "deleteAppDataOnUninstall": true,
      "createDesktopShortcut": true,
      "createStartMenuShortcut": true,
      "shortcutName": "Lucene Index Explorer",
      "artifactName": "LIE.Setup.${version}.${ext}"
    }
  }
```

So if I set `productName` to `LIE` which will generate a `LIE.exe` and the Desktop and Start Menu shortcut shows up as `Lucene Index Explorer` by using `shortcutName` however in the Add or Remove Programs list it displays as LIE as opposed to Lucene Index Explorer.

Is there any config setting I have missed or can use, so I can achieve this please?

This will just make life easier for people viewing the running processes with Task Manager on Windows to see the running LIE.Server process and the main LIE application along side each in the list of processes IMO.

Look forward to hearing back from you.

Thanks,
Warren :)

**Comment by develar** [comment]
I cannot reproduce. And `shortcutName` not required since it is equal to product name.

Please remove old app, restart windows and install again. Still no product name? Attach your exe or send me by email to test and attach here `dist/electron-builder-effective-config.yaml`.

**Comment by warrenbuckley** [comment]
**dist/electron-builder-effective-config.yaml**
```
directories:
  output: dist
  buildResources: build
appId: lucene.index.explorer
productName: Lucene Index Explorer
extraResources:
  from: ../LIE.Server/bin/dist/
  to: LIE.Server/bin/dist/
  filter:
    - '**/*'
win:
  target:
    - nsis
  icon: build/icon.png
nsis:
  deleteAppDataOnUninstall: true
  createDesktopShortcut: true
  createStartMenuShortcut: true
  artifactName: 'LIE.Setup.${version}.${ext}'
electronVersion: 1.8.4
```
So the installer gets generated as `LIE.Setup.1.0.0` just fine
![image](https://user-images.githubusercontent.com/1389894/39036476-8291b4c0-4475-11e8-99d4-d4d50a1d9ba9.png)

However the actual final application exe is called `Lucene Index Explorer.exe` which I want to be named `LIE.exe`

![image](https://user-images.githubusercontent.com/1389894/39036549-c1e6041e-4475-11e8-98c0-765fdbdf333b.png)

And yes shortcut name is fine
![image](https://user-images.githubusercontent.com/1389894/39036770-776ab6fe-4476-11e8-84f9-ce5b14ba3597.png)

So the only problem I have is that the final application .exe is not called `LIE.exe`
![image](https://user-images.githubusercontent.com/1389894/39036873-ca42eafe-4476-11e8-9c74-71620e02cfa2.png)

I attempted to set `productName: "lie"` and also have `shortcutName: "Lucene Index Explorer"` which solves the problem of my final executable being called lie.exe and the desktop shortcut and start menu item being called the full name.

However by doing it this way the list of programs to remove will list/display it as `LIE 1.0.0` and users are unlikely to know to search for that when trying to uninstall the program.

If you need any more details or specifics from me, let me know and thanks again for the awesome help :)

**Comment by skreborn** [comment]
@warrenbuckley I don't think this is currently possible, as `productFileName` (which is later used to identify the actual output file name - the one you are trying to change) is generated based on `productName` using simple sanitization.

If I'm right and this is the case though, it shouldn't be too hard to add the `productFileName` option to the configuration file and only fall back on `productName` if it's not provided.

https://github.com/electron-userland/electron-builder/blob/2dda5fcb45d99a0a991d195a2f577e3dbca775eb/packages/electron-builder-lib/src/appInfo.ts#L32-L33

**Comment by warrenbuckley** [comment]
Feel free to close this and re-open if it its needed.

**Comment by Tanimodori** [comment]
This issue is fixed in #5409, just use `win.executableName` or any other platforms you like.
